services/streammanager/googlesheets/googlesheetsmanager.go

```diff
@@ -0,0 +1,236 @@
+	// log it as parse error for that event
+	keys, values, parseErr := ParseTransformedData(parsedJSON)
+
+	if parseErr != nil {
+		respStatus = "Faliure"
+		responseMessage = "[GoogleSheets] error :: Failed to parse transformed data ::" + parseErr.Error()
+		return 400, respStatus, responseMessage
+
+	}
+	// Converting the string array of keys to an array of interface compatibale with sheets-api (Keys relates to sheet header)
+	headerRow := GetSheetsData(keys)
+	// Creating value range for inserting row into sheet
+	var vr sheets.ValueRange
+	vr.MajorDimension = "ROWS"
+	vr.Range = spreadSheetTab + "!A1"
+	// This is the header row for creating the header in the sheet
+	vr.Values = append(vr.Values, headerRow)
+	_, err := googleAPIService.Service.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
+	if err != nil {
+		statCode, serviceMessage := HandleServiceError(err)
+		respStatus = "Faliure"
+		responseMessage = "[GoogleSheets] error :: Failed to add Header :: " + serviceMessage
+		return statCode, respStatus, responseMessage
+	}
+
+	// In this section we are appending the actual messages into the sheet
+	messageValues := [][]interface{}{}
+	// After parsing the parsedJSON as a row specific data which we are storing in message
+	message := GetSheetsData(values)
+	vr.Values = append(messageValues, message)
+	// Appending the actual message into the specific sheet
+	_, err = googleAPIService.Service.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	if err != nil {
+		statCode, serviceMessage := HandleServiceError(err)
+		respStatus = "Faliure"
+		responseMessage = "[GoogleSheets] error :: Failed to insert Payload :: " + serviceMessage
+		return statCode, respStatus, responseMessage
+	}
+
+	respStatus = "Success"
+	responseMessage = "[GoogleSheets] :: Message Payload inserted with messageId :: " + parsedJSON.Get("id").String()
+	return 200, respStatus, responseMessage
+}
+
+// This method is created for fail safety, if in any case when err type is not googleapi.Error
+// we should not crash with a type error.
+func HandleServiceError(err error) (statusCode int, responseMessage string) {
+	statusCode = 400
+	responseMessage = err.Error()
+	if reflect.TypeOf(err).String() == "*googleapi.Error" {
+		serviceErr := err.(*googleapi.Error)
+		statusCode = serviceErr.Code
+		responseMessage = serviceErr.Message
+	}
+	return statusCode, responseMessage
+}
+
+// source is the json object from transformer and we are iterating the json as a map
+// and we are storing the data into designated position in array based on transformer
+// mappings.
+// Example payload we have from transformer:
+// { message:
+// 	 { 	'0': { key: 'anonymous_id', value: '' },
+// 	 	'1': { key: 'user_id', value: 'userTest004' },
+// 	 	'2': { key: 'event', value: 'Page Call' },
+// 	 	..}
+// }
+func ParseTransformedData(source gjson.Result) ([]string, []string, error) {
```

Comment:
Starting with Capital letter makes a function public. There is no need for this function to be exposed outside.

Reply:
Same with all functions. Don't make the functions public unless needed.

Reply:
Fixed it